crates/topos/tests/cert_delivery.rs

```diff
@@ -309,3 +313,209 @@ async fn cert_delivery() {
+                            positions: vec![]
+                        }),
+                        source_checkpoint: None
+                    }.into()
+                };
+
+                let response = client.watch_certificates(in_stream).await.map_err(|_| {
+                    (
+                        peer_string.clone(),
+                        "Unable to execute the watch_certificates on TCE api",
+                    )
+                })?;
+                let mut resp_stream = response.into_inner();
+                async move {
+                    while let Some(received) = resp_stream.next().await {
+                        let received = received.unwrap();
+                        if let Some(Event::CertificatePushed(CertificatePushed {
+                            certificate: Some(certificate),
+                            ..
+                        })) = received.event
+                        {
+                            // unwrap is safe because we are sure that the certificate is present
+                            if certificate_id == certificate.id.unwrap() {
+                                debug!("Received the certificate on {}", peer_string);
+                                return Ok(());
+                            }
+                        }
+                    }
+
+                    Err((peer_string.clone(), "didn't receive any certificate"))
+                }
+                .await
+            }));
+        }
+
+        let mut client = ApiServiceClient::connect(random_peer.clone())
+            .await
+            .map_err(|_| vec![format!("Unable to connect to the TCE api on {random_peer}")])?;
+
+        // submit a certificate to one node
+        _ = client
+            .submit_certificate(SubmitCertificateRequest {
+                certificate: Some(pushed_certificate.into()),
+            })
+            .await
+            .map_err(|_| {
+                vec![format!(
+                    "Unable to submit the certificate to the TCE api on {random_peer}"
+                )]
+            })?;
+
+        tokio::time::sleep(Duration::from_secs(timeout_broadcast)).await;
+        let mut errors = vec![];
+
+        join_all(join_handlers)
+            .await
+            .iter()
+            .for_each(|result| match result {
+                Err(_) => {
+                    errors.push("Unable to properly execute command".to_string());
+                }
+                Ok(Err((peer, error))) => {
+                    errors.push(format!("{peer} {error}"));
+                }
+                _ => {}
+            });
+
+        if errors.is_empty() {
+            Ok(())
+        } else {
+            Err(errors)
+        }
+    })
+    .await
+    .map_err(|error| {
+        warn!("Timeout reached: {:?}", error);
+        error
+    })
+}
+
+async fn run_check_certificate_delivery(
+    number_of_nodes: usize,
+    timeout_broadcast: u64,
+    timeout: u64,
```

Comment:
Prefer using the Duration type directly from the 1st call

Also, not sure we need to deal with explicit sleeps, cant we just have one high level timeout through the `#[timeout`?